libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -535,6 +535,19 @@ void WaveTrack::DoOnProjectTempoChange(
          clip->OnProjectTempoChange(oldTempo, newTempo);
 }
 
+Finally<std::function<void()>>
+WaveTrack::ProvisionallySynchronizeProjectTempo(WaveTrack& other) const
+{
+   const auto originalTempo = other.GetProjectTempo();
+   const auto provisionalTempo = GetProjectTempo();
+   if (!provisionalTempo.has_value() || !originalTempo.has_value())
+      THROW_INCONSISTENCY_EXCEPTION;
```

Comment:
Now I think, let the exception stand. Rewrite the function as I suggested so it doesn't mutate other but constructs a copy.

If the preconditions are not satisfied, throw. The two places where this function is called are in pasting and I already made sure years ago with exhaustive examination of call paths that these are safe places to throw and have AudacityApp::OnExceptionInMainLoop catch it and deal with it.